forge-ai/src/main/java/forge/ai/ability/TokenAi.java

```diff
@@ -359,4 +372,42 @@ public static Card spawnToken(Player ai, SpellAbility sa) {
         return result;
     }
 
+    private boolean tgtRoleAura(final Player ai, final SpellAbility sa, final Card tok, final boolean mandatory) {
+        boolean isCurse = tok.getFirstAttachSpell().getParamOrDefault("AILogic", "").equals("Curse");
```

Comment:
The SpellAbility that creates the token also should have AILogic Curse

imo that should be checked first